packages/transactions/src/__tests__/decompile-transaction-test.ts

```diff
@@ -316,4 +317,357 @@ describe('decompileTransaction', () => {
+                                1, // nonce authority address
+                            ],
+                            data: new Uint8Array([4, 0, 0, 0]),
+                            programAddressIndex: 3,
+                        },
+                    ],
+                    lifetimeToken: nonce,
+                    staticAccounts: [
+                        // writable signers
+                        feePayer,
+                        // read-only signers
+                        nonceAuthorityAddress,
+                        // writable non-signers
+                        nonceAccountAddress,
+                        // read-only non-signers
+                        systemProgramAddress,
+                        recentBlockhashesSysvarAddress,
+                    ],
+                    version: 0,
+                },
+                signatures: [],
+            };
+
+            const transaction = decompileTransaction(compiledTransaction);
+
+            const expectedInstruction: IInstruction = {
+                accounts: [
+                    {
+                        address: nonceAccountAddress,
+                        role: AccountRole.WRITABLE,
+                    },
+                    {
+                        address: recentBlockhashesSysvarAddress,
+                        role: AccountRole.READONLY,
+                    },
+                    {
+                        address: nonceAuthorityAddress,
+                        role: AccountRole.READONLY_SIGNER,
+                    },
+                ],
+                data: new Uint8Array([4, 0, 0, 0]),
+                programAddress: systemProgramAddress,
+            };
+            expect(transaction.instructions).toStrictEqual([expectedInstruction]);
+        });
+
+        it('converts a durable nonce transaction with multiple instruction', () => {
+            const compiledTransaction: CompiledTransaction = {
+                compiledMessage: {
+                    header: {
+                        numReadonlyNonSignerAccounts: 4, // recent blockhashes sysvar, system program, 2 other program addresses
+                        numReadonlySignerAccounts: 0, // nonce authority already added as fee payer
+                        numSignerAccounts: 1, // fee payer and nonce authority are the same account
+                    },
+                    instructions: [
+                        {
+                            accountIndices: [
+                                1, // nonce account address
+                                3, // recent blockhashes sysvar
+                                0, // nonce authority address
+                            ],
+                            data: new Uint8Array([4, 0, 0, 0]),
+                            programAddressIndex: 2,
+                        },
+                        { programAddressIndex: 4 },
+                        { programAddressIndex: 5 },
```

Comment:
Prob not a big deal, but should be include `accountIndices` here for extra testing?

Reply:
Yep good shout, added some accounts/data to one of the extra instructions. It should be covered by the same code as the conversion of the durable nonce instruction, but that's a bit of an implementation detail so worth adding to another instruction.